### ISSUE TITLE
skip_ansible_lint tag ignored by ansible-lint >= 6.14.1

### ISSUE THREAD

**Issue by ghost** [ansible/ansible-lint#3632]
##### Summary

In recent versions of Ansible lint the tag `skip_ansible_lint` is ignored. I'm currently using v6.17.2.
I tried different versions - this seems to have been introduced between v6.14.0 and v6.14.1.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
ansible-lint 6.17.2 using ansible-core:2.14.5 ansible-compat:4.1.2 ruamel-yaml:0.17.24 ruamel-yaml-clib:0.2.7
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```yaml
---
hosts: localhost
tasks:
  - name: Nginx security
    ansible.builtin.blockinfile:
      path: /etc/nginx/nginx.conf
      insertafter: '# ssl_prefer_server_ciphers'
      block: |
        ssl_protocols TLSv1.2 TLSv1.3;
        ssl_ciphers very long list of ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers;
        other_option on;
        yet_other_option on;
    tags:
      - skip_ansible_lint
```

```console
(venv) user@host ~/s/ansible-new> pip install ansible-lint==6.14.0 > /dev/null && ansible-lint playbooks/playground/test-lint
.yml

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: pip install --upgrade pip

Passed with production profile: 0 failure(s), 0 warning(s) on 1 files.
A new release of ansible-lint is available: 6.14.0 → 6.17.2 Upgrade by running: pip install --upgrade ansible-lint
(venv) user@host  ~/s/ansible-new> pip install ansible-lint==6.14.1 > /dev/null && ansible-lint playbooks/playground/test-lint.yml

[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: pip install --upgrade pip
WARNING  Listing 1 violation(s) that are fatal
yaml[line-length]: Line too long (182 > 160 characters)
playbooks/playground/test-lint.yml:10

Read documentation for instructions on how to ignore specific rule violations.

                Rule Violation Summary
 count tag               profile rule associated tags
     1 yaml[line-length] basic   formatting, yaml

Failed after min profile: 1 failure(s), 0 warning(s) on 1 files.
A new release of ansible-lint is available: 6.14.1 → 6.17.2 Upgrade by running: pip install --upgrade ansible-lint

```

##### Desired Behavior

The task gets skipped by ansible-lint.

##### Actual Behavior

See output above.

**Comment by sf-rag** [comment]
https://github.com/ansible/ansible-lint/pull/3144

> reducing the divergences and user confusions

For me the new behavior is actually more confusing. I think it would be helpful to show an error message that `skip_ansible_lint` has no effect on tasks with YAML formatting violations.

For anyone finding this later, the following works as intended:
```yaml
hosts: localhost
tasks:
# yamllint disable rule:line-length
  - name: Nginx security
    ansible.builtin.blockinfile:
      path: /etc/nginx/nginx.conf
      insertafter: '# ssl_prefer_server_ciphers'
      block: |
        ssl_protocols TLSv1.2 TLSv1.3;
        ssl_ciphers very long list of ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers ciphers;
        other_option on;
        yet_other_option on;
# yamllint enable rule:line-length
```
